Server/ConcordiaCurriculumManager/Controllers/GroupController.cs

```diff
@@ -79,16 +107,76 @@ public async Task<IActionResult> AddUserToGroup(Guid groupId, Guid userId)
     }
 
     [HttpDelete("{groupId}/users/{userId}")]
-    [Authorize(Roles = RoleNames.Admin)]
+    [Authorize]
     [SwaggerResponse(StatusCodes.Status200OK, "User removed from group successfully")]
     [SwaggerResponse(StatusCodes.Status404NotFound, "Error removing user from group")]
     public async Task<IActionResult> RemoveUserFromGroup(Guid groupId, Guid userId)
     {
+        Guid currentUserId;
```

Comment:
This check for authorization can be extracted to a private helper method `IsAdminOrGroupMaster`, since it's repeated in multiple methods.